### PR TITLE
[Decaf] for firefox PR

### PR DESCRIPTION
**Pull request by kuceb**
decaf for #1359 

:-1: |no_squash
-|-
:+1:|image

- [x] wait on #5732 
- [x] DO NOT SQUASH

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -4,5 +4,8 @@
   ],
   "extends": [
     "plugin:@cypress/dev/general"
-  ]
+  ],
```
Comment (Bkucera) [inline .eslintrc.json]: I'll add this into @cypress/dev eslint plugin later

File: packages/server/lib/socket.js [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+  return Promise.delay(25).then(fn)
+}
+
+const isSpecialSpec = (name) => {
+  return name.endsWith('__all')
+}
+
+class Socket {
+  constructor (config) {
+    if (!(this instanceof Socket)) {
```
Comment (Bkucera) [inline packages/server/lib/socket.js]: change this

**Review by flotwig — APPROVED** [review]
yay, amazing :tada::rage1::tada:

**Review by flotwig — CHANGES_REQUESTED** [review]
squash your 4th commit into ya 3rd

**Review by flotwig — APPROVED** [review]
yay, amazing :tada::rage1::tada:

**Pull request merged**